tests/test_utils.py

```diff
@@ -2,6 +2,7 @@
 simple unit tests for this package
 """
 
+import datetime
```

Comment:
Do you need to import `datetime`?

Reply:
You're right.  Thought it would be needed.  Turns out not.

The sort is necessary when using catalogs unpacked from `databroker.pack`.  This was first noticed while preparing lesson materials for upcoming Bluesky case studies training.  Turns out the problem was also happening with the technique to be replaced in #479.  Did not see it until testing for it!

Reply:
`collect_meta` and `show_meta` look interesting.  Can you show example output?  Is it useful for apstools?

Reply:
Thanks for the review.

Reply:
```python
In [13]: show_meta(404, db, 409)
Scan #  motors       scan_type plan_name init. pos. final pos. num_points exit_status
404     fourc_tablex None      rel_scan  -0.5       0.5        40         success    
405     fourc_z      None      rel_scan  -1500      1500       50         success    
406     fourc_tablex None      rel_scan  -0.3       0.3        40         success    
407     fourc_tablex None      rel_scan  -0.3       0.3        40         success    
408     fourc_tablex None      rel_scan  -0.2       0.2        30         success    
409     fourc_tablex None      rel_scan  -0.2       0.2        30         success 

In [14]: show_meta([404, 405, 407, 409], db, meta_keys=['plan_name', 'time'], table_fmt='grid')
+---------+-----------+------------------------------------------------+
| Scan #  | plan_name | time                                           |
+=========+===========+================================================+
| 404     | rel_scan  | ['03/10/2021 16:02:35', '03/10/2021 16:04:38'] |
+---------+-----------+------------------------------------------------+
| 405     | rel_scan  | ['03/10/2021 16:08:36', '03/10/2021 16:11:48'] |
+---------+-----------+------------------------------------------------+
| 407     | rel_scan  | ['03/10/2021 16:20:33', '03/10/2021 16:22:35'] |
+---------+-----------+------------------------------------------------+
| 409     | rel_scan  | ['03/10/2021 17:10:42', '03/10/2021 17:12:14'] |
+---------+-----------+------------------------------------------------+
```


```python
In [19]: meta = collect_meta([404,405,406], db, ['scan_id', 'uid', 'time', 'plan_name'])

In [20]: meta
Out[20]: 
OrderedDict([(404,
              OrderedDict([('scan_id', [404, None]),
                           ('uid',
                            ['707622f5-487d-4e70-9426-a8d63f64fef1',
                             '419e285e-197e-4cbd-985f-aee28ab2f855']),
                           ('time', [1615413755.2195203, 1615413878.4491396]),
                           ('plan_name', ['rel_scan', None])])),
             (405,
              OrderedDict([('scan_id', [405, None]),
                           ('uid',
                            ['8ce4d5dd-2d8f-4264-8bea-4987e566e7f1',
                             '5ecba77a-1dc2-4ed8-8dd1-1bdc3821b0f9']),
                           ('time', [1615414116.6258235, 1615414308.2149165]),
                           ('plan_name', ['rel_scan', None])])),
             (406,
              OrderedDict([('scan_id', [406, None]),
                           ('uid',
                            ['846ee9ce-8640-4a2b-bcd1-e8fa6e308398',
                             '41877e94-bac2-405c-b80a-813d192be947']),
                           ('time', [1615414697.2198677, 1615414816.4029033]),
                           ('plan_name', ['rel_scan', None])]))])
```

I think something like these functions might be useful in apstools. `collect_meta` is more general and will just return the metadata if finds in the start and stop documents. We find it particularly useful to read time, and the hints (which now include information on the monitor, detector, and scan type (in process)). `show_meta` is currently more picky, because it tries to figure out things based on the way we currently collect metadata. For example, we have partially added a "scan_type" to the hints, but a lot of scans still don't have it, so it just returns a useless `None`.

Reply:
This is looking great!  It figures into the strategy of adding _searchable_ metadata to a run so the run can be found later.  There's a symmetry between adding metadata to a run (e.g., `md={"my_key": "blue", "scan_type": "alignment"}`) and searching for match(es) later (e.g., `db.v2.search({"my_key": "blue", "scan_type": "alignment"})`).